core/domain/user_jobs_one_off.py

```diff
@@ -150,6 +153,95 @@ def reduce(userbio_length, stringified_usernames):
+
+        story_ids_linked_to_incomplete_explorations = (
+            list(set(exp_services.get_story_ids_linked_to_explorations(
+                incomplete_exploration_ids))))
+        stories_linked_to_incomplete_explorations = (
+            story_fetchers.get_stories_by_ids(
+                story_ids_linked_to_incomplete_explorations))
+
+        story_ids_linked_to_completed_explorations = (
+            list(set(exp_services.get_story_ids_linked_to_explorations(
+                completed_exploration_ids))))
+        stories_linked_to_completed_explorations = (
+            story_fetchers.get_stories_by_ids(
+                story_ids_linked_to_completed_explorations))
+        topic_ids_linked_to_stories_for_completed_exps = [
+            story.corresponding_topic_id for story in (
+                stories_linked_to_completed_explorations)]
+        topics_linked_to_stories_for_completed_exps = (
+            topic_fetchers.get_topics_by_ids(
+                topic_ids_linked_to_stories_for_completed_exps))
+        completed_story_ids = []
+
+        # Mark stories for completed explorations.
+        for story in stories_linked_to_completed_explorations:
+            if story:
+                completed_nodes = (
+                    story_fetchers.get_completed_nodes_in_story(
+                        user_id, story.id))
+                all_nodes = story.story_contents.nodes
+
+                if len(completed_nodes) != len(all_nodes):
+                    learner_progress_services.record_story_started(
+                        user_id, story.id)
+                else:
+                    learner_progress_services.mark_story_as_completed(
+                        user_id, story.id)
+                    completed_story_ids.append(story.id)
+
+        # Mark stories for incomplete explorations.
+        for story in stories_linked_to_incomplete_explorations:
+            if story:
+                learner_progress_services.record_story_started(
+                    user_id, story.id)
+                if story.corresponding_topic_id:
+                    learner_progress_services.record_topic_started(
+                        user_id, story.corresponding_topic_id)
+
+        # Mark topics for completed explorations.
+        for topic in topics_linked_to_stories_for_completed_exps:
+            if topic:
+                story_ids_in_topic = []
+                for canonical_story in topic.canonical_story_references:
+                    story_ids_in_topic.append(canonical_story.story_id)
+                if set(story_ids_in_topic).intersection(set(
+                        completed_story_ids)):
+                    learner_progress_services.mark_topic_as_learnt(
+                        user_id, topic.id)
+                else:
+                    learner_progress_services.record_topic_started(
+                        user_id, topic.id)
+
+        yield (user_id, 1)
```

Comment:
Sorry, this needs to be 
```suggestion
        yield ('SUCCESS', 1)
```

Reply:
Done.